### DISCUSSION TITLE
Tasmota SML bei PV- und Haushaltszähler

### DISCUSSION THREAD

**Discussion by DasMoritz** [arendst/Tasmota#18126] [General]
Hallo zusammen, 

ich habe eine PV-Anlage und aus alter Zeit noch einen Erzeugungszähler und einen klassischen Haushaltszähler. 
Beide Zähler lese ich via Tasmota mittels SML Anpassungen und IR-Schnittstelle aus, soweit alles fein. 

Nun habe ich jedoch das Problem, dass die MQTT-Datenübermittlung der erfassten Werte zu unterschiedlichen Zeitpunkten erfolgt, damit kann ich meine Berechnungen etwas "knicken", aktuell ist eingestellt dass die Werte alle 10 Sekunden übermittelt werden sollen. 

Dabei entsteht das folgende Problem: 

12:59:55: Erzeugungszähler: -1500W (also Erzeugung)
13:00:00 Haushaltszähler: -500W (also Einspeisung)
Der errechnete Hausverbrauch um 13:00:00 liegt damit bei 1000W 

Nun sinkt bei der nächsten Abfrage die Erzeugungsleistung wegen Wolken, auf sagen wir 750W, damit... 

13:00:05: Erzeugungszähler: -750W (also Erzeugung)
Der errechnete Hausverbrauch um 13:00:05 liegt damit dann bei -250W <<-- Und das kann ja nicht sein. 

Als Trigger für meine interne Berechnung in Blockly (ioBroker) gilt: Sobald Erzeugungsleistung oder Einspeiseleistung aktualisiert wurde, berechne den Hausverbauch. 

Nun die Kernfrage: 
Kann ich die Daten von meinen ESP32 nicht in einem Intervall (alle 10 Sekunden) sondern zu festen Zeitpunkten (wie bei Cron) übermitteln? 

Dann wäre klar, dass beide Daten zu gleichen Zeitpunkten erhoben wurde. 

Danke,
Moritz

**Comment by joba-1** [comment]
Hier ist englisch erwünscht, mit https://www.deepl.com/de/ kannst du bei Bedarf übersetzen :)

either use rules trigger Time#Minute to send something every minute or let your target system send status commands to all tasmota devices at the same time, preferably via mqtt.
